service/access/server.go

```diff
@@ -429,7 +432,8 @@ func (s *Server) ExecuteScriptAtBlockID(ctx context.Context, in *access.ExecuteS
 	blockID := flow.HashToID(in.BlockId)
 	height, err := s.index.HeightForBlock(blockID)
 	if err != nil {
-		return nil, fmt.Errorf("could not get height for block ID %x: %w", blockID, err)
+		err = fmt.Errorf("could not get height for block ID %x: %w", blockID, err)
+		return nil, status.Error(codes.NotFound, err.Error())
```

Comment:
nit: You don't really need to create an error to convert it to a string. here any all the others
```suggestion
		return nil, status.Errorf(codes.NotFound, "could not get height for block ID %x: %v", blockID, err)
```